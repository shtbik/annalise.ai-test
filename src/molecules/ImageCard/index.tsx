import { FC } from 'react'

import { StyledComponent } from '../../types/component'

import { StyledImg } from './styled.index'

interface IImageCard extends StyledComponent {
  url: string
  alt: string
}

const ImageCard: FC<IImageCard> = ({ url, alt, className }) => {
  return (
    <div className={className}>
      <StyledImg src={url} alt={alt} />
    </div>
  )
}

export default ImageCard
