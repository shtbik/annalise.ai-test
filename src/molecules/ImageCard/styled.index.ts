import styled from 'styled-components'

export const StyledImg = styled.img`
  max-width: 100%;
`
