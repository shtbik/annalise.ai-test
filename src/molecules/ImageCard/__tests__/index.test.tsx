import { render } from '@testing-library/react'
import styled from 'styled-components'

import ImageCard from '../'

const SImageCard = styled(ImageCard)`
  filter: blur(5px);
`

const imageUrl =
  'https://upload.wikimedia.org/wikipedia/commons/thumb/b/b6/Image_created_with_a_mobile_phone.png/440px-Image_created_with_a_mobile_phone.png'

describe('ImageCard component', () => {
  it('url passed correctly', () => {
    const { container } = render(<ImageCard url={imageUrl} alt="Avatar" />)
    expect(container.getElementsByTagName('img')[0].src).toEqual(imageUrl)
  })

  it('snapshot', () => {
    const { container } = render(<SImageCard url={imageUrl} alt="Logo" />)
    expect(container).toMatchSnapshot()
  })
})
