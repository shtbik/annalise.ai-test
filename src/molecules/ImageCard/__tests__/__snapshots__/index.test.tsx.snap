// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCard component snapshot 1`] = `
.c1 {
  max-width: 100%;
}

.c0 {
  -webkit-filter: blur(5px);
  filter: blur(5px);
}

<div>
  <div
    class="c0"
  >
    <img
      alt="Logo"
      class="c1"
      src="https://upload.wikimedia.org/wikipedia/commons/thumb/b/b6/Image_created_with_a_mobile_phone.png/440px-Image_created_with_a_mobile_phone.png"
    />
  </div>
</div>
`;
